File: backend/schemas/evaluation.py
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel


class EvaluationResult(BaseModel):
    candidate_name: str
    overall_score: float
    scores: Dict[str, float]
    bonus_points: float
    strengths: List[str]
    improvements: List[str]
    github: Dict[str, Any]
    projects: List[Dict[str, Any]]
    timestamp: datetime


class EvaluationHistoryItem(BaseModel):
    id: int
    candidate_name: str
    overall_score: float
    timestamp: datetime


class EvaluationResponse(BaseModel):
    candidate_name: str
    overall_score: float
    scores: Dict[str, float]
    bonus_points: float
    strengths: List[str]
    improvements: List[str]
    github: Dict[str, Any]
    projects: List[Dict[str, Any]]
    metadata: Optional[Dict[str, Any]] = None

File: frontend/src/types/evaluation.ts
export interface EvaluationScores {
  open_source: number
  self_projects: number
  production: number
  technical_skills: number
}

export interface GithubPayload {
  username?: string
  repositories?: number
  profile?: Record<string, any>
}

export interface EvaluationResponse {
  candidate_name: string
  overall_score: number
  scores: EvaluationScores
  bonus_points: number
  strengths: string[]
  improvements: string[]
  github: GithubPayload
  projects: any[]
}
